kotlintest-assertions-arrow/src/main/kotlin/io/kotlintest/assertions/arrow/try/matchers.kt

```diff
@@ -37,4 +37,20 @@ fun beFailure() = object : Matcher<Try<Any>> {
       is Try.Failure<*> -> Result(true, "Try should be a Failure", "Try should not be Failure")
     }
   }
-}
\ No newline at end of file
+}
+
+inline fun <reified A : Throwable> Try<Any>.shouldBeFailureWithThrowable() = this should beFailureWithThrowable<A>()
+inline fun <reified A : Throwable> Try<Any>.shouldNotBeFailureWithThrowable() = this shouldNot beFailureWithThrowable<A>()
+inline fun <reified A : Throwable> beFailureWithThrowable() = object : Matcher<Try<Any>> {
```

Comment:
What do you think about the more concise name shouldBeFailureOfType ?

Reply:
Yeah that makes sense to me. I'll update the name.